packages/mc-scripts/src/commands/login.js

```
@@ -0,0 +1,98 @@
+const path = require('path');
+const { promisify } = require('util');
+const read = promisify(require('read'));
+const fetch = require('node-fetch');
+const fs = require('fs');
+const homedir = require('os').homedir();
+const { processConfig } = require('@commercetools-frontend/application-config');
+const { MC_API_URLS, DEFAULT_CREDENTIALS } = require('../constants');
+
+function validateEmail(email) {
+  const re = /\S+@\S+\.\S+/;
+  return re.test(email);
+}
+
+const getEmail = async () => {
+  let email = await read({ prompt: 'Email: ' });
+  if (!validateEmail(email)) {
+    console.log('Please input a valid email');
+    email = await getEmail();
+  }
+  return email;
+};
+
+const getPassword = async () => {
+  return await read({ prompt: 'Password: ', silent: true });
+};
+
+const getUpdatedCredentials = (oldCredentials, cloudIdentifier, token) => {
+  oldCredentials[cloudIdentifier] = token;
+  return JSON.stringify(oldCredentials);
+};
+
+const updateCredential = (cloudIdentifier, token) => {
+  const dirpath = `${homedir}/.mcscriptsrc`;
+  const filename = 'credentials.json';
+  const filePath = `${dirpath}/${filename}`;
+
+  try {
+    if (!fs.existsSync(dirpath)) {
+      fs.mkdirSync(dirpath);
+      const credentials = getUpdatedCredentials(
+        DEFAULT_CREDENTIALS,
+        cloudIdentifier,
+        token
+      );
+      console.log({ credentials });
+      fs.writeFileSync(filePath, credentials);
+    } else {
+      const data = fs.readFileSync(filePath, 'utf8');
+      console.log({ data });
+      const credentials = getUpdatedCredentials(
+        JSON.parse(data),
+        cloudIdentifier,
+        token
+      );
+      fs.writeFileSync(filePath, credentials);
+    }
+  } catch (err) {
+    console.error('An error occurred');
+  }
```

No need for this try-catch. Let the error bubble up